### ISSUE TITLE
Rename ECS to Entity-Component Pattern

### ISSUE THREAD

**Issue by kulkalkul** [gdquest-demos/godot-design-patterns#4]
https://www.gdquest.com/tutorial/godot/design-patterns/entity-component-system/

In the post above (and in this repo) ECS used as a misnomer. ECS is a data-oriented pattern, which handled with S, the system part. The example you provide here contains logic in entity and accesses stuff traditional way, which is contradictory with S of ECS.

As an example:

Classical Unity with MonoBehaviour is an example of EC pattern, where each MB is a component with logic and they can be combined into single entity, a.k.a. GameObject.

On the other hand, new Unity DOTS stack has a ECS backend that specifically designed for data-oriented workflow. Both have components for sure, but the main difference between the two is the "runner" part of it, which is a dedicated function that iterates all similar stuff at once (and how data is stored).

So, this is a major disinformation on the internet and I'd be happy to see it fixed in this specific repo, so suggested this. Thanks!

**Comment by NathanLovato** [comment]
I've got to read up on the issue, as I've always seen the Unity community talk about MonoBehavior as an ECS (I used to work with it professionally), but we're certainly not closed to clarifying things. Would you have references to recommend us on the two different patterns?

**Comment by kulkalkul** [comment]
Thanks for the quick reply! Without bombarding with unnecessary information, I have two references I can suggest on the exact topic:

From FLECS's author, linked specific part:
https://github.com/SanderMertens/ecs-faq#how-is-ecs-different-from-entity-component-frameworks

From Aras Pranckevičius, page 44 of the slide mentioned in the tweet, which I also included:
https://twitter.com/aras_p/status/1044656885100675072
http://aras-p.info/texts/files/2018Academy%20-%20ECS-DoD.pdf

**Comment by NathanLovato** [comment]
Thanks, however what i'm looking for is rather evidence that people do not use the expression "entity component system" to talk about the more high level stuff like monobehavior. As mentioned I used to work with Unity way before DOTS and the community referred to Unity using ECS back then too.

I'd like to know if now it's the expectation or meaning of ECS has changed compared to 10 years ago. Not just for individual experienced developers like the two you mentioned, whose work I've seen before, but at large. I.e. would anyone ever search for "EC pattern" or "entity component" without the system part? Because what matters to me is how the target audience expresses themselves and their expectations.

In other words, I'd like to know if users in their majority now expect ECS to be about performance and data-driven rather than developer experience, which is primarily what we cover in this repository, especially as the patterns are implemented in GDScript, an interpreted language, so they can never be fast.

**Comment by kulkalkul** [comment]
Well, the term ECS is not something new and it always meant the same thing. But I bet some developers used ECS interchangeably with EC Framework. Probably some people used word system as in "EC System" instead of framework as they both convey a similar meaning. ECS itself is a abbreviation of Entity Component System, which EC frameworks miss that important "system" part.

I get you want to reach to more people, but the part I care is people searching for real ECS to find your site and read it, thinking that this is how ECS can be done in Godot.

Unity itself clearly uses ECS with right meaning:
https://docs.unity3d.com/Packages/com.unity.entities@0.17/manual/index.html
They mention new DOTS technology as ECS in Unite 2017 conference:
https://youtu.be/GEuT5-oCu_I?t=294
Google results of "unity ecs" before that conference:
![image](https://user-images.githubusercontent.com/1013994/235068341-42845d93-5925-4881-af20-cf09dc9d8b70.png)

https://github.com/sschmid/Entitas
https://github.com/sebas77/Svelto.ECS

Unity ECS Google result filtered for prior 2015:
![chrome_2023-04-28_09-13-48](https://user-images.githubusercontent.com/1013994/235069100-05a599da-df93-49b9-948e-bd4482ae6731.png)

https://gamedev.stackexchange.com/questions/72962/how-does-the-unity3d-entity-system-work/73839

Contrary to that, searching just Entity Component System will show the correct meaning. ECS is new shiny word nowadays, so I can say majority people searching for ECS will expect the correct meaning, not EC framework. There are lots of ECS libraries, frameworks, official game engine backends (Unity ECS), game engines designed specifically for ECS (bevy) out there. I think these are enough.

And, as I said, I get your point. So, if you are still not convinced 😅 after these, atleast consider stating "EC Framework" and "ECS" are different things and what you show is actually an "EC Framework", not an ECS implementation.

**Comment by NathanLovato** [comment]
Sounds good, thanks for taking the time to share more details. We'll update our content. My plate's really full lately but I've added this to my to do list.

**Comment by kulkalkul** [comment]
Today I encountered a similar misconception that describes a similar workflow as "ECS":

https://www.youtube.com/watch?v=ldxDJpRPCMI

It has been a year since I opened this issue, and I strongly believe that the GDQuest website has played a big role in spreading this disinformation towards godot and programming newcomers due to how it is written and how it confuses a lot of facts: talking about EC in the same post without clearly differientiating it, having PowerSystem but also having methods on entities, the demo project that you shared in this repo not being expandable easily (Like, how PowerSystem of yours that "mimics" a system handles entity addition/removal? Because this probably makes people just use EC as in described in your article.), stating that ECS is taxing on update calls, not regarding the last ~8 years of change of that most of ECSs being data oriented. Considering that you have not shown enough concern to address it, and have not devoted even the time I have in communicating with you about a poorly-written article on a website related to your business that promotes your courses, and given that the damage has already been done, I do not want this issue to linger in my account-wide issue tracker for the coming years.

Many thanks, and have a nice day.

**Comment by NathanLovato** [comment]
How about contributing the change?

I maintain over 100 repos, and as I told you last year, my plate's full (and constantly so as I get dozens of people contacting me daily). I appreciate the feedback, and understand this is important to you personally. Our website is open source, so is this repo. It's so people like you can edit things directly when we don't have the time to change them.

**Comment by kulkalkul** [comment]
I've already contributed, creating an issue is contributing. I'd be happy to further contribute and make necessary changes back then but my plate is also full, same as your plate. I have way important things to do than correcting/rewriting guides, which I believe is also the case for you. But the difference is, open-source content on your site has connection to your business, not mine; which I believe makes it your responsibility according to my moral compass.

**Comment by NathanLovato** [comment]
Given that you seem to understand open-source well, and you can empathize as you're in the same situation as I am, I'd have appreciated if you could've just pinged me for a reminder. I never wrote or suggested you hadn't contributed. 

Anyway, this has been on my backlog for a long time, so I'll take a moment to hotfix the article
